### PR TITLE
CMR-8038: Fix CMR search app 500 error when searching with empty JSON

### PR DESCRIPTION
**Pull request by dpesall**
Requests to CMR Search App with an empty JSON should return a 422 `invalid-data` response rather than crashing with a 500 error.

### CODE REVIEW

File: search-app/src/cmr/search/services/json_parameters/conversion.clj [hunk 1]
```diff
@@ -112,6 +112,8 @@
   "Perform all validations against the provided JSON query."
   [concept-type json-query]
   (concept-type-validation concept-type)
+  (when-not json-query
+    (errors/throw-service-errors :invalid-data ["json query cannot be empty"]))
```
[the remark this answers is not in the document]
Comment (dpesall) [reply search-app/src/cmr/search/services/json_parameters/conversion.clj]: Adjusted json -> JSON

File: system-int-test/test/cmr/system_int_test/search/collection_search_format_test.clj [hunk 2]
```diff
@@ -1221,3 +1221,31 @@
...
+                        :connection-manager (s/conn-mgr)})
+            response2 (client/get
+                             (url/search-url :collection)
+                             {:accept "application/json"
+                              :content-type "application/json"
+                              :body ""
+                              :throw-exceptions false
+                              :connection-manager (s/conn-mgr)})]
+         (is (=
+               422
```
Comment (daniel-zamora) [inline system-int-test/test/cmr/system_int_test/search/collection_search_format_test.clj:1245]: should be one line above

File: system-int-test/test/cmr/system_int_test/search/collection_search_format_test.clj [hunk 3]
```diff
@@ -1221,3 +1221,31 @@
...
+               (:status response1)
+               (:status response2)))
+         (is (=
+               "JSON query cannot be empty"
```
Comment (daniel-zamora) [inline system-int-test/test/cmr/system_int_test/search/collection_search_format_test.clj:1249]: should be on line above

File: system-int-test/test/cmr/system_int_test/search/collection_search_format_test.clj [hunk 4]
```diff
@@ -1221,3 +1221,31 @@
...
+    (testing "testing collection search with empty json body"
+      (let [response1 (client/get
+                       (url/search-url :collection)
+                       {:accept "application/json"
+                        :content-type "application/json"
+                        :body nil
+                        :throw-exceptions false
+                        :connection-manager (s/conn-mgr)})
+            response2 (client/get
+                             (url/search-url :collection)
```
Comment (daniel-zamora) [inline system-int-test/test/cmr/system_int_test/search/collection_search_format_test.clj:1238]: fix indent

File: system-int-test/test/cmr/system_int_test/search/collection_search_format_test.clj [hunk 5]
```diff
@@ -1221,3 +1221,31 @@
       (is (= ["STAC result format is only supported for granule searches"]
              (:errors response)
              (:errors ext-response))))))
+
+(deftest search-with-empty-json
+  (let [coll1 (d/ingest "PROV1" (dc/collection))
+        coll-concept-id (:concept-id coll1)]
+    (index/wait-until-indexed)
+    (testing "testing collection search with empty json body"
```
Comment (daniel-zamora) [inline system-int-test/test/cmr/system_int_test/search/collection_search_format_test.clj:1229]: move testing up and combine let

File: system-int-test/test/cmr/system_int_test/search/collection_search_format_test.clj [hunk 6]
```diff
@@ -1221,3 +1221,31 @@
       (is (= ["STAC result format is only supported for granule searches"]
              (:errors response)
              (:errors ext-response))))))
+
+(deftest search-with-empty-json
+  (let [coll1 (d/ingest "PROV1" (dc/collection))
+        coll-concept-id (:concept-id coll1)]
+    (index/wait-until-indexed)
```
Comment (daniel-zamora) [inline system-int-test/test/cmr/system_int_test/search/collection_search_format_test.clj:1228]: can wait until indexed in let by using `_` binding

**Review by daniel-zamora — APPROVED** [review]

**Pull request merged**